torch/_inductor/compile_fx.py

```diff
@@ -1326,6 +1329,9 @@ def compile_fx_aot(
 ) -> Union[List[str], str]:
     assert isinstance(model_, GraphModule), model_
 
+    # [See NOTE] Unwrapping subclasses AOT
+    unwrap_tensor_subclass_parameters(model_)
```

Comment:
cc @angelayi @desertfire. I think this is a case where `aot_compile()` and `ExportedProgram.run_decompositions()` are doing some similar things but they aren't sharing much code, so right now we're duplicating some code between the two (in this case, handling tensor subclass weights during export). Maybe in the future we can think about a way to refactor/share more code?